internal/client/client.go

```diff
@@ -868,6 +868,87 @@ func (a *AuthApiClient) ListResourcePermissions() ybmclient.ApiListResourcePermi
 	return a.ApiClient.AuthApi.ListResourcePermissions(a.ctx)
 }
 
+func (a *AuthApiClient) GetSensitivePermissions() (map[string][]string, error) {
+	resourcePermissionsResp, resp, err := a.ListResourcePermissions().Execute()
+	if err != nil {
+		c, _ := httputil.DumpResponse(resp, true)
+		logrus.Debug(c)
+		return nil, err
+	}
+
+	permissions := resourcePermissionsResp.GetData()
+	sensitivePermissionsMap := map[string][]string{}
+	for _, permission := range permissions {
+		resourceType := string(permission.Info.GetResourceType())
+		for i := 0; i < len(permission.Info.OperationGroups); i++ {
+			operationGroup := string(permission.Info.OperationGroups[i].GetOperationGroup())
+			tags := permission.Info.OperationGroups[i].GetTags()
+			if len(tags) == 0 || tags[0] != "SENSITIVE" {
+				continue
+			}
+			if ops, ok := sensitivePermissionsMap[resourceType]; ok {
+				ops = append(ops, operationGroup)
+				sensitivePermissionsMap[resourceType] = ops
+			} else {
+				sensitivePermissionsMap[resourceType] = []string{operationGroup}
+			}
+		}
+	}
+	return sensitivePermissionsMap, nil
+}
+
+func (a *AuthApiClient) ContainsSensitivePermissions(permissionsMap map[string][]string) bool {
+	sensitivePermissions, err := a.GetSensitivePermissions()
+	if err != nil {
+		return false
+	}
+	for sensitiveResourceType, sensitiveOps := range sensitivePermissions {
+		if ops, ok := permissionsMap[sensitiveResourceType]; ok {
+			for _, op := range ops {
+				for _, sensitiveOp := range sensitiveOps {
+					if op == sensitiveOp {
+						return true
+					}
+				}
+			}
+		}
+	}
+	return false
+}
+
+func (a *AuthApiClient) RoleContainsSensitivePermissions(roleId string) bool {
+	sensitivePermissions, err := a.GetSensitivePermissions()
+	if err != nil {
+		return false
+	}
+
+	roleResp, _, err := a.GetRole(roleId).Execute()
+	if err != nil {
+		return false
```

Comment:
I think you should log the error here or better return the error and handle in the caller.

Reply:
Done